cluster/gce/gci/configure-helper.sh

```diff
@@ -811,20 +811,26 @@ function assemble-docker-flags {
     # If using a network plugin, extend the docker configuration to always remove
     # the network checkpoint to avoid corrupt checkpoints.
     # (https://github.com/docker/docker/issues/18283).
-    echo "Extend the default docker.service configuration"
+    echo "Extend the docker.service configuration to remove the network checkpiont"
     mkdir -p /etc/systemd/system/docker.service.d
     cat <<EOF >/etc/systemd/system/docker.service.d/01network.conf
 [Service]
 ExecStartPre=/bin/sh -x -c "rm -rf /var/lib/docker/network"
 EOF
+  fi
 
-    systemctl daemon-reload
+  # COS m60 sets the default TasksMax that is too low for us. Bump it up.
+  # (https://github.com/kubernetes/kubernetes/issues/51977)
+  echo "Extend the docker.service configuration to set a higher pids limit"
+  mkdir -p /etc/systemd/system/docker.service.d
+  cat <<EOF >/etc/systemd/system/docker.service.d/02tasksmax.conf
+[Service]
+TasksMax=10000
```

Comment:
Can we make this even bigger, something like 1048600?

Reply:
The current number 4915 is the "15% of "max number of tasks on the system" according to the COS team. How about setting it to `infinity` (i.e., no limit)?